arch-chroot.in

```diff
@@ -74,19 +75,41 @@ chroot_add_resolv_conf() {
   chroot_add_mount "$src" "$dest" --bind
 }
 
-while getopts ':hNu:' flag; do
+arch-chroot() {
+  (( EUID == 0 )) || die 'This script must be run with root privileges'
+
+  [[ -d $chrootdir ]] || die "Can't create chroot on non-directory %s" "$chrootdir"
+
+  $setup "$chrootdir" || die "failed to setup chroot %s" "$chrootdir"
+  if (( keepresolvconf )); then
+    chroot_add_resolv_conf "$chrootdir" || die "failed to setup resolv.conf"
+  fi
```

Comment:
The logic seems inverted?

Reply:
:facepalm: indeed - original used "changeresolvconf". Thanks fixed.